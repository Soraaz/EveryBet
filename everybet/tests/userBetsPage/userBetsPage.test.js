import React from "react";
import {NavigationEvents} from "react-navigation";
import renderer from "react-test-renderer";
import {shallow} from "enzyme/build";


import UserBetsPage from "../../src/userBetsPage/userBetsPage";

describe("UserBetsPage", () => {
	const navigation = {
		navigate: jest.fn(x => x),
		state: {params: {type: "type"}}
	};

	describe("rendering", () => {
		it("Should render correctly", () => {
			const tree = renderer
				.create(<UserBetsPage navigation={navigation}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () => {
		let wrapper;

		beforeEach(() => {
			wrapper = shallow(<UserBetsPage navigation={navigation}/>);
		});

		it("Should call the callback increasing the close state", () => {
			expect(wrapper.instance().state.close).to.be.equal(0);
			wrapper.find(NavigationEvents).first().props().onWillFocus();
			expect(wrapper.instance().state.close).to.be.equal(1);
		});
	});
});