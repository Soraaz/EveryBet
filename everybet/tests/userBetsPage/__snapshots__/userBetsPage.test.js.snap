// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBetsPage rendering Should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": "#37597F",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#37597F",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={
        Object {
          "testUri": "../../../assets/images/everybet_logo.png",
        }
      }
      style={
        Object {
          "height": 39.25,
          "marginLeft": 25,
          "width": 138,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 20,
          "justifyContent": "flex-start",
          "marginLeft": "auto",
          "width": 90,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/betcoin_icon.png",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "System",
            "fontSize": 10,
            "marginLeft": "auto",
            "marginRight": 5,
            "textAlign": "right",
          }
        }
      >
        0
      </Text>
    </View>
    <Image
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={
        Object {
          "testUri": "../../../assets/images/user_placeholder.png",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": 40,
          "marginHorizontal": 15,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
        },
        Object {
          "backgroundColor": "#F4F4FE",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#37597F",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginHorizontal": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Catégorie
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 5,
              "height": 25,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "paddingLeft": 10,
              }
            }
          >
            <Text>
              Toutes
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginHorizontal": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Ordre
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 5,
              "height": 25,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "paddingLeft": 10,
              }
            }
          >
            <Text>
              Plus récents
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
